### PR TITLE
allow file:/// resources

### PR DESCRIPTION
**Pull request by clburlison**
Allow support for file:/// type resources. This was tested with dmgs, packages and the configuration plist so we should be good. 

Was too simple for me to not make this change earlier.

### CODE REVIEW

**Comment by grahamgilbert** [comment]
Good so far. Rather than printing your errors to stdout, would you be able to get them displaying like the other imagr errors?

**Comment by clburlison** [comment]
Alright. I can work on that. I also found out I didn't actually test packages properly. So additional changes are needed.

**Comment by grahamgilbert** [comment]
Excellent :)

Sent from my iPhone

On Thu, Feb 25, 2016 at 10:15 AM -0800, "Clayton Burlison" notifications@github.com wrote:

Alright. I can work on that. I also found out I didn't actually test packages properly. So additional changes are needed.

**Comment by clburlison** [comment]
Alright, if I didn't screw anything up this is good to go. I'm sending error messages for packages/dmgs to the Imagr error message screen. As it stands scripts are using the generic, "Couldn't copy script XX" message. I wasn't 100% sure how to tie the http error codes in for that section.

If anyone wants to give this a spin here is a sample imagr_config.plist

<< examples located in the docs >> https://github.com/clburlison/imagr/wiki/Creating-an-External-Drive

Working on documentation next.

**Comment by grahamgilbert** [comment]
Nice one, I'll try and have a play tomorrow.

File: Imagr/Utils.py [hunk 1]
```diff
@@ -517,14 +536,29 @@ def unmountdmg(mountpoint):
...
+        source = url_parse.path
+        try:
+            if os.path.isfile(source):
+                shutil.copy(source, file)
+            else:
+                shutil.copytree(source, file)
+            return source, None
+        except:
+            error = "Unable to copy %s" % url
+            return None, error
```
Comment (grahamgilbert) [inline Imagr/Utils.py]: Shouldn't this return False?

**Comment by clburlison** [comment]
So I'm running into an issue with COSXIP and local files. Need to find physical hardware to see why the installer is failing. I started the documentation and can merge whenever: https://github.com/clburlison/imagr/wiki/Creating-an-External-Drive

**Comment by grahamgilbert** [comment]
I was going to suggest hdiutil wasn't liking the `file://` url, but I just tested that and it mounted fine. Sorry, can't be of more use.

**Comment by clburlison** [comment]
I'm an idiot. All is well. I was attempting to escape a path and python wouldn't give me a good error message until I tired it on a physical machine. 

As far as I am concerned this PR is ready to be merged. Lets hold off a few days as I've got two people testing it (or they say they are).

I will need help with merging the Creating-an-External-Drive doc to the wiki as it has pictures embed. The only way GH wiki will embed pictures is with a git push so you'd need to do the merge yourself or give me temporary commit access to this repo.

**Comment by grahamgilbert** [comment]
Add the pictures in somewhere with this PR and I'll merge it before the release next week.

**Comment by clburlison** [comment]
- Images have been added to this commit.
- [Documentation updated](https://github.com/clburlison/imagr/wiki/Creating-an-External-Drive) (I'll update the 'real' wiki after this PR is merged).
- Validateplist updated for file:/// urls. 
- I made one more minor change to how we're dealing with URLs. This fixes an issue with spaces in the URL. https://github.com/grahamgilbert/imagr/pull/128/commits/31f860f7514c5f22dc7d943161b3d415af097a60

> Due to how Imagr is passing to `installer` and `hdiutil` we can't treat all file:/// resources the same. As such, we mandate that all local resources have spaces escaped as %20 then manually remove the %20 when passing to `installer` and when copying the files via `shutil.copy`.

The above commit really is only for NBICreator as it forces you to use "Image Volume" as the name of your external drive when booted into the NetInstall Environment. A slight limitation but pretty easy to work around.

I've also passed all tests I could think of.

**Comment by grahamgilbert** [comment]
Ok, let's do this thing. Merging.

**Pull request merged**